assets/js/components/settings/settings-modules.js

```diff
@@ -91,6 +92,10 @@ class SettingsModules extends Component {
 				} );
 				return;
 			}
+
+			// Clears session and local storage on every successful setting.
+			clearAppLocalStorage();
```

Comment:
Agreed with @aaemnnosttv, let's call this only in `modulePromise.then`, and in addition in the `! modulePromise` clause before (since not having a module promise wouldn't necessarily be an error).